person.rb

```diff
@@ -0,0 +1,30 @@
+require_relative 'main'
+
+class Person < Nameable
+  attr_accessor :name, :age
+  attr_reader :id
+
+  def initialize(id, age, name = 'Unknown', parent_permission: true)
+    super()
+    @id = id
```

Comment:
[OPTIONAL]
- [ ] Kindly for a better user experience it's not safe to give the end user the responsibility to set the `id` by themselves, so you may consider generating it when an object is instantiated and remove the `id` parameter
Hint:
```rb
id = Random.rand(1...1000) #for example
```
- [ ] As said early, you may consider removing the `super` keyword as the parent class Nameable doesn't have any parameter in the `initialize` method

Reply:
Okay thank you